### PR TITLE
Fix #1675: Implement {Integer,Long}.{highest,lowest}OneBit.

### LINKED ISSUE #1675: Add {Integer, Long}.lowestOneBit and Long.highestOneBit
The methods `Integer.lowestOneBit, Long.lowestOneBit, Long.highestOneBit` are missing.

I will do a PR for this myself, as I need these for spire port.

### PR DESCRIPTION
**Pull request by sjrd**
`Integer.highestOneBit` was already there. This commit completes the set.

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/IntegerTest.scala [hunk 1]
```diff
@@ -169,6 +169,15 @@ object IntegerTest extends JasmineTest {
       expect(Integer.highestOneBit(Integer.MAX_VALUE)).toEqual(0x40000000)
     }
 
+    it("should provide `lowestOneBit`") {
+      expect(Integer.lowestOneBit(0)).toEqual(0)
+      expect(Integer.lowestOneBit(-1)).toEqual(1)
+      expect(Integer.lowestOneBit(-256)).toEqual(256)
+      expect(Integer.lowestOneBit(12)).toEqual(4)
+      expect(Integer.lowestOneBit(0x88)).toEqual(0x8)
+      expect(Integer.lowestOneBit(Int.MaxValue)).toEqual(1)
```
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/IntegerTest.scala]: Should also add the test for `Int.MinValue`

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala [hunk 2]
```diff
@@ -185,6 +185,26 @@ object LongTest extends JasmineTest {
       expect(JLong.toString(0x7fffffffffffffffL, 16)).toEqual("7fffffffffffffff")
     }
 
+    it("should provide `highestOneBit`") {
+      expect(JLong.highestOneBit(0L) == 0L).toBeTruthy
+      expect(JLong.highestOneBit(-1L) == Long.MinValue).toBeTruthy
+      expect(JLong.highestOneBit(-256L) == Long.MinValue).toBeTruthy
+      expect(JLong.highestOneBit(1L) == 1L).toBeTruthy
+      expect(JLong.highestOneBit(0x88L) == 0x80L).toBeTruthy
+      expect(JLong.highestOneBit(Long.MaxValue) == 0x4000000000000000L).toBeTruthy
```
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala]: Should also add the test for `Long.MinValue`

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala [hunk 3]
```diff
@@ -185,6 +185,26 @@ object LongTest extends JasmineTest {
...
+    }
+
+    it("should provide `lowestOneBit`") {
+      expect(JLong.lowestOneBit(0L) == 0L)
+      expect(JLong.lowestOneBit(-1L) == 1L)
+      expect(JLong.lowestOneBit(-256L) == 256L)
+      expect(JLong.lowestOneBit(12L) == 4L)
+      expect(JLong.lowestOneBit(0x88L) == 0x8L)
+      expect(JLong.lowestOneBit(Long.MaxValue) == 1L)
+      expect(JLong.lowestOneBit(0x32100012300L) == 0x100L).toBeTruthy
```
Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala]: Same `MinValue`

**Comment by nicolasstucki** [comment]
Thats all, otherwise LGTM

**Comment by gzm0** [comment]
Hahahahahaha, it seems my pedantry about tests indeed propagates very fast :)

**Comment by sjrd** [comment]
retest this please

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala [hunk 4]
```diff
@@ -185,6 +185,28 @@ object LongTest extends JasmineTest {
...
+      expect(JLong.highestOneBit(-256L) == Long.MinValue).toBeTruthy
+      expect(JLong.highestOneBit(1L) == 1L).toBeTruthy
+      expect(JLong.highestOneBit(0x88L) == 0x80L).toBeTruthy
+      expect(JLong.highestOneBit(Long.MaxValue) == 0x4000000000000000L).toBeTruthy
+      expect(JLong.highestOneBit(Long.MinValue) == Long.MinValue).toBeTruthy
+      expect(JLong.highestOneBit(0x32100012300L) == 0x20000000000L).toBeTruthy
+    }
+
+    it("should provide `lowestOneBit`") {
+      expect(JLong.lowestOneBit(0L) == 0L)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala]: `.toBeTruthy` is missing all over here.

**Comment by sjrd** [comment]
Updated.

**Comment by nicolasstucki** [comment]
Still LGTM

**Pull request merged**